streams/src/test/java/org/apache/kafka/streams/processor/internals/RepartitionTopicsTest.java

```diff
@@ -162,14 +169,17 @@ public void shouldThrowMissingSourceTopicException() {
             "[test] "
         );
 
-        assertThrows(MissingSourceTopicException.class, repartitionTopics::setup);
+        assertThat(repartitionTopics.setup(), equalTo(false));
```

Comment:
Could you also verify that in the cases without missing source topics `setup()` returns `true` in the other unit tests?